src/config.rs

```diff
@@ -45,9 +52,12 @@ impl ConfigBuilderExt for ConfigBuilder<ClientConfig, WantsVerifier> {
             valid_count,
             invalid_count
         );
-        assert!(!roots.is_empty(), "no CA certificates found");
+        if roots.is_empty() {
+            crate::log::debug!("no native root CA certificates found");
+            None?
```

Comment:
and this can return something more useful, -- i suggest including the logged message, and the `valid_count` and `invalid_count` diagnostics too (i appreciate there's logging going on, but it's less immediate than returning the same stuff in a `Result::Err`, even if it gets smushed into a String along the way)